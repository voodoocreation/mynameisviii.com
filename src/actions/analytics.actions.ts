import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("ANALYTICS");

export const trackEvent = createAction<{
  event: string;
  [index: string]: any;
}>("TRACK_EVENT");
