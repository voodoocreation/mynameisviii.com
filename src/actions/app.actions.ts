import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("APP");

export const initApp = createAction.async<{ locale?: string }, {}>("INIT");

export const toggleNavigation = createAction("TOGGLE_NAVIGATION");

export const setOnlineStatus = createAction<boolean>("SET_ONLINE_STATUS");

export const setHasNewVersion = createAction<boolean>("SET_HAS_NEW_VERSION");

export const setCurrentRoute = createAction<string>("SET_CURRENT_ROUTE");

export const changeRoute = createAction.async<string, {}>("CHANGE_ROUTE");
