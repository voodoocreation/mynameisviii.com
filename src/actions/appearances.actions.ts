import actionCreatorFactory from "typescript-fsa";

import { IAppearance, IDynamoResponse, ILatLng } from "../models/root.models";

const createAction = actionCreatorFactory("APPEARANCES");

export const fetchAppearances = createAction.async<
  {},
  IDynamoResponse<IAppearance, "startingAt">
>("FETCH");

export const fetchMoreAppearances = createAction.async<
  {},
  IDynamoResponse<IAppearance, "startingAt">
>("FETCH_MORE");

export const fetchAppearanceBySlug = createAction.async<string, IAppearance>(
  "FETCH_BY_SLUG"
);

export const setCurrentAppearanceSlug = createAction<string>(
  "SET_CURRENT_SLUG"
);

export const geocodeCurrentAppearanceAddress = createAction.async<{}, ILatLng>(
  "GEOCODE_CURRENT_ADDRESS"
);
