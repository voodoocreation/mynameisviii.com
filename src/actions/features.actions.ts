import actionCreatorFactory from "typescript-fsa";

const actionCreator = actionCreatorFactory("FEATURES");

export const addFeatures = actionCreator<PLAddFeatures>("ADD");

export const removeFeatures = actionCreator<PLRemoveFeatures>("REMOVE");
