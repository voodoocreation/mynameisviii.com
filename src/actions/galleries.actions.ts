import actionCreatorFactory from "typescript-fsa";

const actionCreator = actionCreatorFactory("GALLERIES");

export const fetchGalleries = actionCreator.async<
  {},
  PLFetchGalleriesDone,
  PLFetchGalleriesFailed
>("FETCH");

export const fetchMoreGalleries = actionCreator.async<
  {},
  PLFetchGalleriesDone,
  PLFetchGalleriesFailed
>("FETCH_MORE");

export const fetchGalleryBySlug = actionCreator.async<
  PLFetchGalleryBySlugStarted,
  PLFetchGalleryBySlugDone,
  PLFetchGalleryBySlugFailed
>("FETCH_BY_SLUG");

export const setCurrentGallerySlug = actionCreator<PLSetCurrentGallerySlug>(
  "SET_CURRENT_SLUG"
);
