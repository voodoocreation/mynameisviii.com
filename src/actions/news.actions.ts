import actionCreatorFactory from "typescript-fsa";

const actionCreator = actionCreatorFactory("NEWS");

export const fetchLatestNews = actionCreator.async<
  {},
  PLFetchLatestNewsDone,
  PLFetchLatestNewsFailed
>("FETCH");

export const fetchMoreLatestNews = actionCreator.async<
  {},
  PLFetchLatestNewsDone,
  PLFetchLatestNewsFailed
>("FETCH_MORE");

export const fetchNewsArticleBySlug = actionCreator.async<
  PLFetchNewsArticleBySlugStarted,
  PLFetchNewsArticleBySlugDone,
  PLFetchNewsArticleBySlugFailed
>("FETCH_BY_SLUG");

export const setCurrentNewsArticleSlug = actionCreator<
  PLSetCurrentNewsArticleSlug
>("SET_CURRENT_SLUG");
