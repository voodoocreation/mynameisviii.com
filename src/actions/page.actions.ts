import actionCreatorFactory from "typescript-fsa";

const actionCreator = actionCreatorFactory("PAGE");

export const toggleNavigation = actionCreator<{}>("TOGGLE_NAVIGATION");

export const changeRoute = actionCreator.async<
  PLChangeRouteStarted,
  PLChangeRouteDone,
  PLChangeRouteFailed
>("CHANGE_ROUTE");
