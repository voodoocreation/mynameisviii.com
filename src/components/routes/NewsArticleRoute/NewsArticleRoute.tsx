import Head from "next/head";
import * as React from "react";
import { WrappedComponentProps } from "react-intl";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch } from "redux";

import { absoluteUrl } from "../../../helpers/dataTransformers";
import injectIntlIntoPage from "../../../helpers/injectIntlIntoPage";
import { INewsArticle } from "../../../models/root.models";
import { TStoreState } from "../../../reducers/root.reducers";
import { IPageContext } from "../../connected/App/App";
import Loader from "../../presentation/Loader/Loader";
import NewsArticle from "../../presentation/NewsArticle/NewsArticle";

import * as actions from "../../../actions/root.actions";
import * as selectors from "../../../selectors/root.selectors";

interface IProps extends WrappedComponentProps {
  article?: INewsArticle;
  fetchNewsArticleBySlug: typeof actions.fetchNewsArticleBySlug.started;
  isLoading: boolean;
}

class NewsArticleRoute extends React.Component<IProps> {
  public static async getInitialProps(context: IPageContext) {
    const { query, store } = context;
    const slug = query.slug as string;

    store.dispatch(actions.setCurrentNewsArticleSlug(slug));

    const state = store.getState();

    if (!selectors.getCurrentNewsArticle(state)) {
      store.dispatch(actions.fetchNewsArticleBySlug.started(slug));
    }
  }

  public render() {
    const { article, isLoading } = this.props;
    const { formatMessage } = this.props.intl;

    if (isLoading) {
      return <Loader className="PageLoader" />;
    }

    if (!article) {
      return null;
    }

    return (
      <>
        <Head>
          <title>
            {article.title}
            {" Â· "}
            {formatMessage({ id: "BRAND_NAME" })}
          </title>

          <meta content={article.excerpt} name="description" />

          <meta property="og:title" content={article.title} />
          <meta property="og:description" content={article.excerpt} />
          <meta
            property="og:url"
            content={absoluteUrl(`/news/${article.slug}`)}
          />
          <meta property="og:type" content="article" />
          <meta property="article:published_time" content={article.createdAt} />
          <meta property="article:author" content={article.author} />
          <meta property="og:image" content={article.ogImageUrl} />
          <meta property="og:image:width" content="1200" />
          <meta property="og:image:height" content="630" />
        </Head>

        <NewsArticle {...article} />
      </>
    );
  }
}

const mapState = (state: TStoreState) => ({
  article: selectors.getCurrentNewsArticle(state),
  isLoading: selectors.getNewsIsLoading(state)
});

const mapActions = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      fetchNewsArticleBySlug: actions.fetchNewsArticleBySlug.started
    },
    dispatch
  );

export default injectIntlIntoPage(
  connect(mapState, mapActions)(NewsArticleRoute)
);
