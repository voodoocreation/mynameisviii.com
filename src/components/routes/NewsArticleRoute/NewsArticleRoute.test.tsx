import * as actions from "../../../actions/root.actions";
import { newsArticle } from "../../../models/root.models";
import ComponentTester from "../../../utilities/ComponentTester";
import MockPageContext from "../../../utilities/MockPageContext";
import NewsArticleRoute from "./NewsArticleRoute";

const item = newsArticle({ slug: "test-1" });

const defaultState = {
  news: {
    currentSlug: item.slug,
    items: {
      [item.slug]: item
    }
  }
};

const component = new ComponentTester(NewsArticleRoute, true);

describe("[routes] <NewsArticleRoute />", () => {
  describe("getInitialProps", () => {
    const context = new MockPageContext()
      .withDefaultQuery({
        slug: item.slug
      })
      .withDefaultReduxState(defaultState);

    describe("when the article isn't already in the store", () => {
      const slug = "new-slug";

      it("calls getInitialProps method", async () => {
        await NewsArticleRoute.getInitialProps(
          context.withQuery({ slug }).toObject()
        );
      });

      it("dispatches actions.setCurrentNewsArticleSlug with expected payload", () => {
        const matchingActions = context.reduxHistory.filter(
          actions.setCurrentNewsArticleSlug.match
        );

        expect(matchingActions).toHaveLength(1);
        expect(matchingActions[0].payload).toBe(slug);
      });

      it("dispatches actions.fetchNewsArticleBySlug.started", () => {
        expect(
          context.reduxHistory.filter(
            actions.fetchNewsArticleBySlug.started.match
          )
        ).toHaveLength(1);
      });
    });

    describe("when the article is already in the store", () => {
      it("calls getInitialProps method", async () => {
        await NewsArticleRoute.getInitialProps(context.toObject());
      });

      it("dispatches actions.setCurrentNewsArticleSlug", () => {
        expect(
          context.reduxHistory.filter(actions.setCurrentNewsArticleSlug.match)
        ).toHaveLength(1);
      });

      it("doesn't dispatch actions.fetchNewsArticleBySlug.started", () => {
        expect(
          context.reduxHistory.filter(
            actions.fetchNewsArticleBySlug.started.match
          )
        ).toHaveLength(0);
      });
    });
  });

  it("matches snapshot", () => {
    const { wrapper } = component.withReduxState(defaultState).mount();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it("renders a loader when isLoading is true", () => {
    const { wrapper } = component
      .withReduxState({
        ...defaultState,
        news: { isLoading: true }
      })
      .mount();

    expect(wrapper.find("Loader")).toHaveLength(1);
  });

  it("doesn't render anything when no article exists", () => {
    const { wrapper } = component
      .withReduxState({
        news: {
          currentSlug: undefined,
          hasError: true,
          items: {}
        }
      })
      .mount();

    expect(wrapper.render().html()).toBeNull();
  });
});
