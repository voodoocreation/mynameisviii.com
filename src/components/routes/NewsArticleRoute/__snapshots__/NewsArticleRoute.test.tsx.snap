// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[routes] <NewsArticleRoute /> renders 404 error page when no news article exists 1`] = `
<article
  class="ErrorPage"
>
  <header
    class="PageHeader"
  >
    <h1>
      Page not found
    </h1>
  </header>
  <div
    class="ErrorPage-content"
  >
    <svg
      attr="[object Object]"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <p>
      This page does not exist - please use the site navigation to find what you're looking for.
    </p>
  </div>
</article>
`;

exports[`[routes] <NewsArticleRoute /> renders 500 error page when slug isn't in the store 1`] = `
<article
  class="ErrorPage"
>
  <header
    class="PageHeader"
  >
    <h1>
      An error has occurred
    </h1>
  </header>
  <div
    class="ErrorPage-content"
  >
    <svg
      attr="[object Object]"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <p>
      An error has occurred, please try again.
    </p>
  </div>
</article>
`;

exports[`[routes] <NewsArticleRoute /> renders a loader when item is being fetched 1`] = `
<span
  class="Loader PageLoader"
>
  <div
    class="Loader-animation"
  >
    <span />
    <span />
    <span />
  </div>
</span>
`;

exports[`[routes] <NewsArticleRoute /> renders correctly 1`] = `
<article
  class="NewsArticle"
>
  <header
    class="PageHeader"
  >
    <h1>
      New website launched
    </h1>
  </header>
  <div
    class="NewsArticle-body"
  >
    <figure
      class="Image NewsArticle-image isLoading"
    >
      <div
        class="Image-wrapper"
      >
        <img
          alt="New website launched"
          src="https://s3.amazonaws.com/mynameisviii-static/news/new-website-launched.jpg"
        />
        <span
          class="Loader"
        >
          <div
            class="Loader-animation"
          >
            <span />
            <span />
            <span />
          </div>
        </span>
      </div>
    </figure>
    <section
      class="NewsArticle-meta"
    >
      <div
        class="NewsArticle-posted"
      >
        <svg
          attr="[object Object]"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
         
        <span>
          Posted
        </span>
         
        <time
          class="DateTime"
          datetime="2018-06-20T21:20:00"
          title="Wednesday, June 20, 2018, 9:20 PM"
        >
          <span>
            in 6 months
          </span>
        </time>
         
        <span>
          by
        </span>
         Viii
      </div>
      <div
        class="Type Type-news"
      >
        <svg
          attr="[object Object]"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M23 12l-2.44-2.78.34-3.68-3.61-.82-1.89-3.18L12 3 8.6 1.54 6.71 4.72l-3.61.81.34 3.68L1 12l2.44 2.78-.34 3.69 3.61.82 1.89 3.18L12 21l3.4 1.46 1.89-3.18 3.61-.82-.34-3.68L23 12zm-10 5h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
         News
      </div>
    </section>
    <section
      class="NewsArticle-content"
    >
      <div
        class="NewsArticle-contentText"
      >
        <p>
          The brand new official Viii website is now live!
        </p>
        <p>
          The new site features a fully responsive design for viewing on all mobile devices, as well as desktop browsers, plus an all-new news section, releases section and an appearances section for upcoming shows.
        </p>
        <p>
          Stay tuned for more features to be released in the near future.
        </p>
      </div>
    </section>
  </div>
</article>
`;
